scripts/globals/pets/wyvern.lua

```diff
@@ -123,8 +123,13 @@ function onMobSpawn(mob)
     master:addListener("EXPERIENCE_POINTS", "PET_WYVERN_EXP", function(player, exp)
         local pet = player:getPet()
         local prev_exp = pet:getLocalVar("wyvern_exp")
+        -- cap exp at 1000 to prevent wyvern leveling up many times from large exp awards
+        local currentExp = exp
```

Comment:
first, you'd want to move this block inside the prev_exp < 1000 block. then, you'd want to check exp + prev_exp > 1000 (instead of just exp), and then instead of setting exp to 1000 if it is, set exp to 1000 - prev_exp.  this would guarantee that exp is capped to the remaining exp before the wyvern is level capped